File: sem2/2.cpp
#include<iostream>
using namespace std;
int delele(int arr[], int n, int x)
{

   if (arr[n-1] == x)
       return (n-1);
 
   int prev = arr[n-1], i;
   for (i=n-2; i>=0 && arr[i]!=x; i--)
   {
       int curr = arr[i];
       arr[i] = prev;
       prev = curr;
   }
 
   if (i < 0)
     return 0;
 
   arr[i] = prev;
 
   return (n-1);
}
 
int main()
{
    int n, r;
    cout << "size of array : ";
    cin >> n;
    int v[n];
    for(int i=0;i<n;i++)
        cin >> v[i];
    cout<<"element to be deleted :";
    cin >> r; 

    n = delele(v, n, r);
 
    cout << "new arr: ";
    for (int i=0; i<n; i++)
       cout << v[i] << " ";
 
    return 0;
}

File: sem2/2.py
def deleteElement(arr,n,x):
    if arr[n-1]==x:
        return n-1
 
    prev = arr[n-1]
    for i in range(n-2,1,-1):
        if arr[i]!=x:
            curr = arr[i]
            arr[i] = prev
            prev = curr
 
    if i<0:
        return 0
    arr[i] = prev
    return n-1

arr = [11,15,6,8,9,10]
n = len(arr)
x = 6
n = deleteElement(arr,n,x)
print("Modified array is")
for i in range(n):
    print(arr[i],end=" ")